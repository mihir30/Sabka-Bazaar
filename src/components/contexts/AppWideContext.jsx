import React, { createContext, useReducer } from "react";
import { reducer } from "../reducers/ReducerContext";

const initialState = {
  count: 0,
  products: {},
};

export const AppWideContext = createContext(initialState);

const AppWideContextProvider = ({ children }) => {
  const [cartItems, dispatch] = useReducer(reducer, initialState);

  return (
    <AppWideContext.Provider value={{ cartItems, dispatch }}>
      {children}
    </AppWideContext.Provider>
  );
};

export default AppWideContextProvider;
