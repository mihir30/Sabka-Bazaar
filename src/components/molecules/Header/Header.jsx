import React from "react";
import "./Header.scss";
import logo from "../../../static/images/logo.png";
import Cart from "../Cart/Cart";
import { useLocation } from "react-router-dom";
import { Link } from "react-router-dom";
import { Image } from "../../atoms";
import useForm from "../../../validations/validateRegister";
import HamBurgerMenu from "./HamburgerMenu/HamBurgerMenu";

const Header = () => {
  const { getEmail, removeEmail } = useForm();
  const location = useLocation();
  let searchQuery = location.pathname;
  let data = <div></div>;
  let logout = <div></div>;
  if (getEmail() === null) {
    data = <div></div>;
  } else {
    if (searchQuery === "/login") {
      data = <div></div>;
      logout = <div></div>;
      removeEmail();
    } else if (searchQuery === "/register") {
      data = <div></div>;
      logout = <div></div>;
      removeEmail();
    } else {
      data = (
        <div>
          Hello, <span className="header-email">{getEmail()}</span>
        </div>
      );
      logout = (
        <Link to="/login" className="nav-login-links links">
          Logout
        </Link>
      );
    }
  }

  return (
    <header className="header" aria-label="header">
      <div className="header-box">
      <section className="wrapper">
          <HamBurgerMenu />
        </section>
        <Link to="/">
          <Image alt="sabka bazaar" source={logo} className="logo" />
        </Link>
        
        <nav className="header-nav">
          <Link to="/" className="nav-links links">
            Home
          </Link>
          <Link to="/products/" className="nav-links links">
            Products
          </Link>
        </nav>
        <div className="hello-message-wrapper">
          <span className="hello-message">{data}</span>
          <div className="header-nav-wrapper">
            <nav className="header-nav-login">
              <Link to="/login" className="nav-login-links links">
                SignIn
              </Link>
              <Link to="/register" className="nav-login-links links">
                Register
              </Link>
              {logout}
            </nav>

            <Cart />
          </div>
        </div>
      </div>
    </header>
  );
};
export default Header;
