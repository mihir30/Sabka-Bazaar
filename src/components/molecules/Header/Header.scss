.header {
    width: 100%;
    box-shadow: 0px 0px 15px 0px #9e9e9e;
}

.header-box {
    display: flex;
    justify-content: space-between;
    flex-direction: row;
    margin: 0 auto;
    font-size: large;
    font-weight: bold;
    width: 100%;
}

.nav-links,
.nav-login-links {
    text-decoration: none;
    color: #4d4d4d;
}

.nav-links {
    margin: 0 0rem 0 1.8rem;
}

.nav-login-links {
    margin: 0.8rem;
}

.links:hover {
    color: #cc0044;
}


.logo{
    display: none;
}
.header-nav,
.header-nav-login{
    display: none;
}
.hello-message {
    display: block;
}
.header-email{
    color: #cc0044;
    margin-right: 15px;
}
.wrapper{
    display: flex;
}
@media screen and (min-width: 768px) {
    .wrapper{
        display: none;
    }
    .logo {
        display: block;
        width: 6.5rem;
        height: 3.3rem;
        margin-top: 0.4rem;
        cursor: pointer;
    }
    .header-nav {
        display: flex;
        flex-grow: 1;
        align-items: flex-end;
        margin-left: 2.5rem;
        padding-right: 4.5rem;
        padding-bottom: 0.5rem;
    }

    .header-nav-login {
        display: inline-block;
        margin-left: 1.5rem;
    }

    .header-nav-wrapper {
        min-height: 100%;
        display: flex;
        flex-direction: column;
        justify-content: space-between;
    }

    .logo {
        width: 7.5rem;
        height: 4.3rem;
    }
    .hello-message{
        display: inline;
        padding-bottom: 0.5rem;
    }
    
    .hello-message-wrapper{
        display: flex;
        flex-direction: row;
        align-items: flex-end; 
               
    }
}

@media screen and (min-width: 1024px) {
    .header-box {
        padding: 0rem 6rem;
    }
    .logo {
        width: 8.5rem;
        height: 4.3rem;
    }
      
}