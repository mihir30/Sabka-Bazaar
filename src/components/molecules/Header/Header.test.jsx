import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { AppWideContext } from '../../contexts/AppWideContext';
import ReactDom from "react-dom";
import Header from './Header';

describe("Header Component", () => {

    it("should render the component properly", () => {
        const headerEl = document.createElement("header");
        ReactDom.render(<AppWideContext.Provider value={{ cartItems: { count: 0 } }}><BrowserRouter><Header/></BrowserRouter></AppWideContext.Provider>, headerEl); 
    });

    it('should have the right header component snapshot', () => {
        const tree = renderer.create(<AppWideContext.Provider value={{ cartItems: { count: 0 } }}><BrowserRouter><Header /></BrowserRouter></AppWideContext.Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    })
});