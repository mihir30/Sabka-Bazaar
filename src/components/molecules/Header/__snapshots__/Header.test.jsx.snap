// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should have the right header component snapshot 1`] = `
<header
  aria-label="header"
  className="header"
>
  <div
    className="header-box"
  >
    <section
      className="wrapper"
    >
      <section
        className="menu"
      >
        <div
          className="hamburgerMenu"
          onClick={[Function]}
        >
          <img
            alt="Menu Collapsed"
            className="menuCollapsed"
            height={32}
            src="https://cdn.iconscout.com/icon/free/png-256/hamburger-menu-462145.png"
            width={32}
          />
        </div>
        <nav
          className="dropdownWrap "
        >
          <ul
            onClick={[Function]}
          >
            <ul
              className="navigation"
            >
              <li>
                <a
                  activeclassname="active"
                  aria-current="page"
                  className="menuItem active"
                  href="/"
                  onClick={[Function]}
                >
                  HOME
                </a>
              </li>
              <li>
                <a
                  activeclassname="active"
                  className="menuItem"
                  href="/products"
                  onClick={[Function]}
                >
                  PRODUCTS
                </a>
              </li>
              <li>
                <a
                  activeclassname="active"
                  className="menuItem"
                  href="/login"
                  onClick={[Function]}
                >
                  SIGN IN
                </a>
              </li>
              <li>
                <a
                  activeclassname="active"
                  className="menuItem"
                  href="/register"
                  onClick={[Function]}
                >
                  REGISTER
                </a>
              </li>
            </ul>
          </ul>
        </nav>
      </section>
    </section>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="sabka bazaar"
        className="logo"
        data-testid="image"
        src="logo.png"
      />
    </a>
    <nav
      className="header-nav"
    >
      <a
        className="nav-links links"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <a
        className="nav-links links"
        href="/products/"
        onClick={[Function]}
      >
        Products
      </a>
    </nav>
    <div
      className="hello-message-wrapper"
    >
      <span
        className="hello-message"
      >
        <div />
      </span>
      <div
        className="header-nav-wrapper"
      >
        <nav
          className="header-nav-login"
        >
          <a
            className="nav-login-links links"
            href="/login"
            onClick={[Function]}
          >
            SignIn
          </a>
          <a
            className="nav-login-links links"
            href="/register"
            onClick={[Function]}
          >
            Register
          </a>
          <div />
        </nav>
        <div
          className="cart-btn"
        >
          <a
            className="nav-cart-items"
            href="/cartItems"
            onClick={[Function]}
          >
            <button
              className="btn-cart-button"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="cart"
                className="cart"
                data-testid="image"
                src="cart.svg"
              />
              <span
                className="cart-quantity"
                data-testid="count-items"
              >
                0 items
              </span>
            </button>
          </a>
        </div>
        <div
          className="cart-modal-overlay"
        >
          <div
            className="cart-modal"
          >
            <div
              className="minicart-modal-background"
            >
              <div
                className="minicart-empty"
              >
                <div
                  className="cart-quantity-tile"
                >
                  <p>
                    <strong
                      data-testid="my-cart-text"
                    >
                      My Cart 
                    </strong>
                  </p>
                </div>
                <div
                  className="cart-quantity-tile-modal"
                >
                  <p>
                    <strong
                      data-testid="my-cart-text-modal"
                    >
                      My Cart 
                    </strong>
                  </p>
                  <i
                    className="fas fa-times"
                    id="close-btn-empty-cart"
                    onClick={[Function]}
                  />
                </div>
                <div>
                  <p
                    className="empty-cart-text"
                    data-testid="no-items"
                  >
                    No items in your cart
                  </p>
                  <p
                    className="empty-cart-desc"
                    data-testid="fav-items"
                  >
                    Your favourite items are just a click away
                  </p>
                </div>
                <div
                  className="checkout-empty-tile"
                >
                  <div>
                    <button
                      className="btn-checkout"
                      onClick={[Function]}
                    >
                      <span
                        className="start-shopping-btn"
                      >
                        <span>
                          Start Shopping
                        </span>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
